src/gui/DatabaseOpenWidget.cpp

```diff
@@ -137,6 +137,14 @@ void DatabaseOpenWidget::load(const QString& filename)
     clearForms();
 
     m_filename = filename;
+    if (m_filename.contains(".old.")) {
+        auto* msgBox = new QMessageBox();
+        msgBox->setWindowTitle(tr("Opening a backup file"));
+        msgBox->setText(tr("You are opening a database backup. Please be aware that it could contain old entries."));
```

Comment:
I would rephrase it to

> It seems that you are opening a backup database, which may not be up to date.
Do you want to open the File.kdbx instead? Yes - No

It should be paired with a check a) whether there is a file with the same name, but without old and b) whether that file has a newer date.

Reply:
`[Open File] [Cancel]` is confusing. Does Open File open the new file or the old? Does Cancel open the old one, or not open anything? What if the user actually does want to open the old file?
I suggest instead `[Open Old] [Open New] [Cancel]`, where cancel doesn't open anything.

Reply:
One more thought: maybe display a warning banner inside the main window while a backup is open. Similar to the "unstable version" warning.